src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -443,7 +445,9 @@ class Config:
                     with (await asyncio.wait_for(asyncssh.connect(
                             remote_host,
                             port=remote_port,
-                            username=remote_username), timeout=5)) as conn:
+                            username=remote_username,
+                            client_keys=set(glob.glob(search)) - set(glob.glob(exclude_from_search))
```

Comment:
We are using exact same statement above. Can we put it in a variable so we can reuse and avoid extra computation?